src/naive/time/mod.rs

```diff
@@ -565,6 +570,161 @@ impl NaiveTime {
         (time, -rhs) // safe to negate, rhs is within +/- (2^63 / 1000)
     }
 
+    /// Adds given `TimeDelta` to the current time,
+    /// and also returns the number of *seconds*
+    /// in the integral number of days ignored from the addition.
+    /// (We cannot return `TimeDelta` because it is subject to overflow or underflow.)
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use chrono::{Days, NaiveTime};
+    /// use core::time::Duration;
+    ///
+    /// let from_hms = NaiveTime::from_hms;
+    ///
+    /// assert_eq!(from_hms(3, 4, 5).checked_overflowing_add(Duration::from_secs(60 * 60 * 11)),
+    ///            Some((from_hms(14, 4, 5), Days::new(0))));
+    /// assert_eq!(from_hms(3, 4, 5).checked_overflowing_add(Duration::from_secs(60 * 60 * 23)),
+    ///            Some((from_hms(2, 4, 5), Days::new(1))));
+    /// ```
+    pub fn checked_overflowing_add(&self, rhs: Duration) -> Option<(NaiveTime, Days)> {
```

Comment:
I've renamed to `overflowing_add_opt` but perhaps there is a better option.

The original intention was to use `checked` because of the `Option` and `overflowing` represents the fact that it is ok for the `Duration` to be longer than the time to the end of the day.